UART wrapper (chrysn's cleanup)

chrysn opened the pull request:
This PR just contains fixes relative to https://github.com/RIOT-OS/rust-riot-wrappers/pull/39; let's keep discussion there, but it's a PR so that we can see what CI says.

Teufelchen1 commented on src/uart.rs:
```diff
@@ -0,0 +1,355 @@
+    _phantom: core::marker::PhantomData<&'cb ()>,
+}
+
+impl<'cb> UartDevice<'cb> {
```
Why is this block separated from the identical `impl<'cb> UartDevice<'cb>` in line 246?

chrysn replied:
Because I kept the order of functions rather than keeping impls together. In other words historical reasons, hysterical raisins, no reason at all.

Shuffling them around (along with others for better documentation structure); not sure where to best put that during squashing.

chrysn commented:
I've made a bit of a mess with which branch to push to, and GitHub's view of when things were pushed is inconsistent -- it shows "Use From for error conversion" before your review, but actually I don't think you've seen them. I don't think they are major, but maybe you'd like to revisit (best viewed individually)

* a4818924595f334e20274595a1da6a6104b9e5b2
* f749c35416e3f767be9d7136220091242e5fb339
* 4432771481e20597c544bd5fd25c672524c9facd + 9c3186d69d788152eb53df631ec90967227b3abb
* c819fdc30b61b31ece58d6fb1817f6c4c38b1a02

and the recent 9e052afd9dc4d60f3e1b58c1b444c627ccbe9924 (but that's just sequence changes)

I'll do some squashing, trying to strike a balance between documenting the genesis and keeping small fixes out of the way; I'll force-push the tree-identical result.

chrysn commented:
Squashed now into what I think is a sensible sequence of ocmmits, leaning towards keeping history.

* GitHub is said to be a bit annoying when @-mentioning users in commit messages; removed the @.
* Applied `cargo fmt` to all commits using `git filter-branch -f --tree-filter 'cargo fmt; git add **/*.rs' fa66f7f0eec46679bf6c031f1d2588068c323972..` (This has no impact )
* Squashed fixes where they are simple oversights
* Squashed groups of commits where there was no significant difference. Note that for applying reviewers' comments, there is a batch feature in some (but not all) GitHub views of a PR
* Reordered commits to reduce the chances of build failures (even though I don't expect to ever have a bisect hit in here, given that as a whole the PR just introduces a feature)

chrysn commented:
As a last-minute removal I had to ditch the `get_[rt]x_pin` methods -- those are just not available consistently in RIOT across boards; it's a niche feature probably only tested wherever anyone had a concrete use case and implemented it on the boards, and this would be the first time it is being checked for across the boards.

Enabling auto-merge already; that will fail anyway if there is any follow-up trouble (which I don't expect).

kbarning commented:
Sorry for not answering, I was very busy during the last weeks. Is there anything left to do @chrysn?

chrysn commented:
Sure, that was a time period where one is bound to get mixed availabilities.

I think all is done here :-)

kbarning commented:
Alright, thank you for all your time and effort :)
